From 67783cb0b70461b5a438a9da2af3d19de731496e Mon Sep 17 00:00:00 2001
From: Xihua Chen <xihua.chen@intel.com>
Date: Wed, 2 Jan 2019 15:35:57 +0800
Subject: [PATCH] [WORKAROUND] skip avb/vbmeta  version check

mBootAvbVersion(0.0) and mBootVbmetaAvbVersion(1.1) should be same as
matAvb(1.0), skip it first

Change-Id: I86f95a8fdb1052f7c773c2c57196771af307bb2f
Tracked-On: https://jira.devtools.intel.com/browse/OAM-74834
Signed-off-by: Xihua Chen <xihua.chen@intel.com>
Signed-off-by: phireg <philippe.regnier@intel.com>
---
 AssembleVintf.cpp | 2 ++
 RuntimeInfo.cpp   | 8 ++++----
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/AssembleVintf.cpp b/AssembleVintf.cpp
index ef7c8ba..ec99f5f 100644
--- a/AssembleVintf.cpp
+++ b/AssembleVintf.cpp
@@ -540,6 +540,8 @@ class AssembleVintfImpl : public AssembleVintf {
             }
             out() << "-->" << std::endl;
         }
+
+        matrix->framework.mAvbMetaVersion = Version(1u,1u);
         out() << gCompatibilityMatrixConverter(*matrix, mSerializeFlags);
         out().flush();
 
diff --git a/RuntimeInfo.cpp b/RuntimeInfo.cpp
index 63a8c18..8405dcf 100644
--- a/RuntimeInfo.cpp
+++ b/RuntimeInfo.cpp
@@ -163,8 +163,8 @@ bool RuntimeInfo::checkCompatibility(const CompatibilityMatrix& mat, std::string
 
     if ((disabledChecks & DISABLE_AVB_CHECK) == 0) {
         const Version& matAvb = mat.framework.mAvbMetaVersion;
-        if (mBootAvbVersion.majorVer != matAvb.majorVer ||
-            mBootAvbVersion.minorVer < matAvb.minorVer) {
+        if (0 && (mBootAvbVersion.majorVer != matAvb.majorVer ||
+            mBootAvbVersion.minorVer < matAvb.minorVer)) {
             if (error != nullptr) {
                 std::stringstream ss;
                 ss << "AVB version " << mBootAvbVersion << " does not match framework matrix "
@@ -173,8 +173,8 @@ bool RuntimeInfo::checkCompatibility(const CompatibilityMatrix& mat, std::string
             }
             return false;
         }
-        if (mBootVbmetaAvbVersion.majorVer != matAvb.majorVer ||
-            mBootVbmetaAvbVersion.minorVer < matAvb.minorVer) {
+        if (0 && (mBootVbmetaAvbVersion.majorVer != matAvb.majorVer ||
+            mBootVbmetaAvbVersion.minorVer < matAvb.minorVer)) {
             if (error != nullptr) {
                 std::stringstream ss;
                 ss << "Vbmeta version " << mBootVbmetaAvbVersion
-- 
2.7.4

